core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -72,6 +92,66 @@
+         builder.allowBlockingCallsInside(UDP.class.getName(), "_send");
+         builder.allowBlockingCallsInside(FlowControl.class.getName(), "adjustCredit");
+         // Random JGroups stuff
+         builder.allowBlockingCallsInside(Table.class.getName(), "add");
+         // Just assume all the thread pools don't block - not we check the rejected handler below though
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "getTask");
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "processWorkerExit");
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "execute");
+         builder.allowBlockingCallsInside(ScheduledThreadPoolExecutor.class.getName(), "schedule");
+
+         // TODO: This can actually block us for a bit, we should really remove this!!!
+         builder.allowBlockingCallsInside(StateTransferLockImpl.class.getName(), "acquireSharedTopologyLock");
+         builder.allowBlockingCallsInside(StateTransferLockImpl.class.getName(), "releaseSharedTopologyLock");
+
+         // LimitedExecutor just submits a task to another thread pool
+         builder.allowBlockingCallsInside(LimitedExecutor.class.getName(), "execute");
+
+         // Some test utilities may block the cpu thread - which is fine
+         builder.allowBlockingCallsInside(EvictionWithConcurrentOperationsTest.class.getName() + "$Latch", "blockIfNeeded");
+         builder.allowBlockingCallsInside(CheckPoint.class.getName(), "await");
+         builder.allowBlockingCallsInside(CheckPoint.class.getName(), "trigger");
+         builder.allowBlockingCallsInside(BlockingInterceptor.class.getName(), "blockIfNeeded");
+         builder.allowBlockingCallsInside(TestingUtil.class.getName(), "sleepRandom");
+         builder.allowBlockingCallsInside(TestingUtil.class.getName(), "sleepThread");
+
+         // The blocking iterator locks to signal at the end - ignore
+         builder.allowBlockingCallsInside(BlockingFlowableIterable.class.getName() + "$BlockingFlowableIterator", "signalConsumer");
+
+         // Allow logging to block
+         builder.allowBlockingCallsInside(Logger.class.getName(), "logf");
+         builder.allowBlockingCallsInside(java.util.logging.Logger.class.getName(), "log");
+
+         // Make sure our rejection handlers never block in a non blocking thread
+         builder.disallowBlockingCallsInside(BlockingRejectedExecutionHandler.class.getName(), "rejectedExecution");
+         builder.disallowBlockingCallsInside(NonBlockingRejectedExecutionHandler.class.getName(), "rejectedExecution");
+
+         builder.blockingMethodCallback(bm -> {
+            throw new CacheException(String.format("Blocking call! %s on thread %s", bm, Thread.currentThread()));
+         });
+      }).install();
+
+      Thread.setDefaultUncaughtExceptionHandler((thread, t) -> {
+         t.printStackTrace();
```

Comment:
Unfortunately there is no great way to catch these types of errors. We have places where we ignore the error from a CompletionStage or Future. So this is attempting to print out those messages.

Reply:
I don't think we check the console output manually that often, so I'd suggest crafting a JUnit report here too.
Also it's enough to just log the exception with FATAL level, (part of) the stack trace will be printed on the console.

Reply:
Yeah, I was just being overly cautious :)

I will definitely look at the report.